### PR TITLE
adopt new `divviup-client`

### PR DESCRIPTION
**Pull request by tgeoghegan**
The PRs related to https://github.com/divviup/divviup-api/issues/542 implement the new concept of "collector credential". This entails changes to the API for provisioning tasks via divviup-api that we use in e2e integration tests. In this commit we adopt the new `divviup-client` and fix up its usage.

Part of #2071

### CODE REVIEW

**Comment by tgeoghegan** [comment]
Possibly more changes incoming here as I continue to e2e test against divviup-api. We'll also need to backport this to `release/0.5`.

File: integration_tests/Cargo.toml [hunk 1]
```diff
@@ -15,7 +15,8 @@ in-cluster = ["dep:k8s-openapi", "dep:kube"]
 anyhow.workspace = true
 backoff = { version = "0.4", features = ["tokio"] }
 base64.workspace = true
-divviup-client = { git = "https://github.com/divviup/divviup-api", features = ["admin"], rev = "06667dbeb93d4870bb257b7ddf9a19b37a7e5da5" }
+# TOOD(timg): change pinned commit or tag once https://github.com/divviup/divviup-api/pull/565 lands
+divviup-client = { git = "https://github.com/divviup/divviup-api", features = ["admin"], rev = "523e797c85c7aeaca39f91a90f96165e64046e86" }
```
Comment (jbr) [inline integration_tests/Cargo.toml:19]:
```suggestion
divviup-client = { git = "https://github.com/divviup/divviup-api", features = ["admin"], rev = "4f63e553c6b62f8578df3c9369e0010fbccbe0dc" }
```

**Comment by jbr** [comment]
We probably want to cut a divviup-api release and use the version tag for this cli pr

**Comment by tgeoghegan** [comment]
> We probably want to cut a divviup-api release and use the version tag for this cli pr

Good point, and we need to cut a divviup-api release anyway to pick up the changes to address impedance mismatches with the Janus aggregator API. https://github.com/divviup/divviup-api/pull/592

**Review by jbr — APPROVED** [review]
LGTM aside from the tag

**Review by divergentdave — APPROVED** [review]

**Pull request merged**